$OpenBSD: patch-src_route_c,v 1.4 2021/07/02 10:27:11 sthen Exp $

From 39b397dddda51d45958ced162cf1c7a589f65866 Mon Sep 17 00:00:00 2001
From: Roy Marples <roy@marples.name>
Date: Mon, 18 Jan 2021 11:31:05 +0000
Subject: [PATCH] script: Use rt_proto_add to ensure no duplicate interfaces on
 OpenBSD

OpenBSD allows matching priorities, so we need to take the interfaces
in the order given to ensure uniqueness.

Index: src/route.c
--- src/route.c.orig
+++ src/route.c
@@ -409,6 +409,7 @@ rt_proto_add_ctx(rb_tree_t *tree, struct rt *rt, struc
 		return rt;
 
 	rt_free(rt);
+	errno = EEXIST;
 	return NULL;
 }
 
